<launch>
  <arg name="map_file"  default="corridor" />

  <!--  ************** Global Parameters ***************  -->
  <param name="/use_sim_time" value="true" />

  <!--  ************** Stage Simulator *****************  -->
  <node pkg="stage_ros_add_pose_and_crash" type="stageros" name="stageros" args="$(find teb_local_planner_tutorials)/stage/$(arg map_file).world" />

  <!--  ******************* Maps ***********************  -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(find teb_local_planner_tutorials)/maps/$(arg map_file).yaml" output="screen">
    <param name="frame_id" value="map"/>
  </node>

  <arg name="robot_name" default="robot_0" />

  <!--  ************** Navigation ROBOT 0 *************  -->
  <include file="$(find my_patrol_sim)/launch/includes/one_robot.launch.xml">
    <arg name="robot_name" value="robot_0" />
  </include>

  <!-- ****************** Obstacles ******************** -->
  <group ns="robot_1">
    <param name="tf_prefix" value="robot_1"/>
    <node name="Mover" pkg="teb_local_planner_tutorials" type="move_obstacle.py" output="screen"/>
    <param name="vel_y" value="0.3" />
  </group>

  <group ns="robot_2">
    <param name="tf_prefix" value="robot_2"/>
    <node name="Mover" pkg="teb_local_planner_tutorials" type="move_obstacle.py" output="screen"/>
    <param name="vel_y" value="0.3" />
  </group>

  <group ns="robot_3">
    <param name="tf_prefix" value="robot_3"/>
    <node name="Mover" pkg="teb_local_planner_tutorials" type="move_obstacle.py" output="screen"/>
    <param name="vel_y" value="0.3" />
  </group>

  <group ns="robot_4">
    <param name="tf_prefix" value="robot_4"/>
    <node name="Mover" pkg="teb_local_planner_tutorials" type="move_obstacle.py" output="screen"/>
    <param name="vel_y" value="0.3" />
  </group>

  <group ns="robot_5">
    <param name="tf_prefix" value="robot_5"/>
    <node name="Mover" pkg="teb_local_planner_tutorials" type="move_obstacle.py" output="screen"/>
    <param name="vel_y" value="0.3" />
  </group>

  <group ns="robot_6">
    <param name="tf_prefix" value="robot_6"/>
    <node name="Mover" pkg="teb_local_planner_tutorials" type="move_obstacle.py" output="screen"/>
    <param name="vel_y" value="0.3" />
  </group>

  <group ns="robot_7">
    <param name="tf_prefix" value="robot_7"/>
    <node name="Mover" pkg="teb_local_planner_tutorials" type="move_obstacle.py" output="screen"/>
    <param name="vel_y" value="0.3" />
  </group>

  <group ns="robot_8">
    <param name="tf_prefix" value="robot_8"/>
    <node name="Mover" pkg="teb_local_planner_tutorials" type="move_obstacle.py" output="screen"/>
    <param name="vel_y" value="0.3" />
  </group>

  <group ns="robot_9">
    <param name="tf_prefix" value="robot_9"/>
    <node name="Mover" pkg="teb_local_planner_tutorials" type="move_obstacle.py" output="screen"/>
    <param name="vel_y" value="0.3" />
  </group>

  <group ns="robot_10">
    <param name="tf_prefix" value="robot_10"/>
    <node name="Mover" pkg="teb_local_planner_tutorials" type="move_obstacle.py" output="screen"/>
    <param name="vel_y" value="0.3" />
  </group>

  <group ns="robot_11">
    <param name="tf_prefix" value="robot_11"/>
    <node name="Mover" pkg="teb_local_planner_tutorials" type="move_obstacle.py" output="screen"/>
    <param name="vel_y" value="0.3" />
  </group>

  <group ns="robot_12">
    <param name="tf_prefix" value="robot_12"/>
    <node name="Mover" pkg="teb_local_planner_tutorials" type="move_obstacle.py" output="screen"/>
    <param name="vel_y" value="0.3" />
  </group>

  <group ns="robot_13">
    <param name="tf_prefix" value="robot_13"/>
    <node name="Mover" pkg="teb_local_planner_tutorials" type="move_obstacle.py" output="screen"/>
    <param name="vel_y" value="0.3" />
  </group>

  <node name="ground_truth_obstacles" pkg="teb_local_planner_tutorials" type="publish_ground_truth_obstacles.py" output="screen" />

  <!--node name="visualize_velocity_profile" pkg="teb_local_planner_tutorials" type="visualize_velocity_profile.py" output="screen" /-->

  <!--  **************** Visualisation ****************  -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find teb_local_planner_tutorials)/cfg/rviz_navigation_cc.rviz">
    <remap from="/move_base_simple/goal" to="/robot_0/move_base_simple/goal" />
  </node>

</launch>
